src/btree/col_srch.c

```diff
@@ -254,6 +254,14 @@ __wt_col_search(
         if ((cip = __col_var_search(current, recno, NULL)) == NULL) {
             cbt->recno = __col_var_last_recno(current);
             cbt->slot = page->entries == 0 ? 0 : page->entries - 1;
+            if (cbt->recno != WT_RECNO_OOB) {
```

Comment:
It deserve a comment here why we need to search again.

Reply:
Thanks, fixed in 26089bb.